internal/cryptsetup/cryptsetup.go

```diff
@@ -50,9 +51,12 @@ var (
 
 // CryptSetup manages encrypted devices.
 type CryptSetup struct {
-	nameInit func(name string) (cryptDevice, error)
-	pathInit func(path string) (cryptDevice, error)
-	device   cryptDevice
+	nameInit                 func(name string) (cryptDevice, cryptDevice, string, string, error)
+	pathInit                 func(path string) (cryptDevice, cryptDevice, string, string, error)
+	device                   cryptDevice
+	deviceWithAttachedHeader cryptDevice
+	headerDevice             string
+	headerFile               string
```

Comment:
This could maybe use some comments on what the individual device handles and names are